embedded/document/engine.go

```diff
@@ -24,122 +24,142 @@ import (
 
 	"github.com/codenotary/immudb/embedded/sql"
 	"github.com/codenotary/immudb/embedded/store"
+	"github.com/codenotary/immudb/pkg/api/protomodel"
 
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
 const (
-	DocumentIDField   = "_id"
-	DocumentBLOBField = "_doc"
+	DefaultDocumentIDField = "_id"
+	DocumentBLOBField      = "_doc"
 )
 
 var (
 	ErrIllegalArguments       = store.ErrIllegalArguments
+	ErrUnsupportedType        = errors.New("unsupported type")
 	ErrCollectionDoesNotExist = errors.New("collection does not exist")
 	ErrMaxLengthExceeded      = errors.New("max length exceeded")
 	ErrMultipleDocumentsFound = errors.New("multiple documents found")
 	ErrDocumentNotFound       = errors.New("document not found")
 	ErrDocumentIDMismatch     = errors.New("document id mismatch")
 	ErrNoMoreDocuments        = errors.New("no more documents")
+	ErrFieldAlreadyExists     = errors.New("field already exists")
+	ErrFieldDoesNotExist      = errors.New("field does not exist")
+	ErrReservedFieldName      = errors.New("reserved field name")
+	ErrLimitedIndexCreation   = errors.New("index creation is only supported on empty collections")
 )
 
-// Schema to ValueType mapping
+func mayTranslateError(err error) error {
```

Comment:
let's call it translateSQLError, or wrapSQLError or anything as such. Use of the word "may" makes it a doubtful clause when reading the code and is not explicit

Reply:
so translateErr is fine